src/components/menu/examples/menu-subitems-lazy-loading.tsx

```diff
@@ -0,0 +1,74 @@
+import {
+    MenuItem,
+    ListSeparator,
+    LimelMenuCustomEvent,
+} from '@limetech/lime-elements';
+import { Component, State, h } from '@stencil/core';
+
+/**
+ * Lazy loading infinite amount of sub menu items.
+ */
+@Component({
+    tag: 'limel-example-menu-subitems-lazy-loading',
+    shadow: true,
+})
+export class MenuSubItemsLazyLoadingExample {
+    private items: Array<MenuItem | ListSeparator> = [
+        {
+            text: 'Item 1',
+        },
+        {
+            text: 'Item 2',
+        },
+        {
+            text: 'Item 3',
+        },
+        {
+            text: 'Item 4',
+        },
+        {
+            text: 'Item 5',
+        },
+    ];
+
+    @State()
+    private lastSelectedItem: string;
+
+    public render() {
+        return [
+            <limel-menu
+                lazyLoadItems
+                loadSubItems={this.handleLoadSubItems}
+                items={this.items}
+                onSelect={this.handleSelect}
+            >
+                <limel-button label="Menu" slot="trigger" />
+            </limel-menu>,
+            <limel-example-value
+                label="Last selected item"
+                value={this.lastSelectedItem}
+            />,
+        ];
+    }
+
+    private handleLoadSubItems = async (
+        item: MenuItem
+    ): Promise<MenuItem[]> => {
+        return new Promise<MenuItem[]>((resolve) => {
+            setTimeout(() => {
+                const subItems = [];
+                for (let i = 1; i < 6; i++) {
+                    subItems.push({
+                        text: `${item.text}.${i}`,
+                    });
+                }
+
+                resolve(subItems);
+            }, 1000);
```

Comment:
We've had some trouble with people blindly copy-pasting example code into production, even though you would think it's been made pretty obvious:

<img width="542" alt="image" src="https://github.com/Lundalogik/lime-elements/assets/154725/50f8d899-404c-4be3-af70-be93dcf08f31">

So please try to make it a little harder to miss that this timeout is there for _demo purposes only_. 😅

(See the _With no suggestions and a message for empty search results_ example for `limel-picker`, but also feel free to come up with something better than what we have there.)

Reply:
fixed